src/main/java/org/sonarqube/gradle/SourceCollector.java

```diff
@@ -101,19 +112,27 @@ public SourceCollector(Set<Path> existingSources, Set<Path> directoriesToIgnore,
 
   @Override
   public FileVisitResult preVisitDirectory(Path path, BasicFileAttributes basicFileAttributes) throws IOException {
-    if (
-      isHidden(path) ||
-      isExcludedDirectory(path) ||
-      isCoveredByExistingSources(path)
-    ) {
+    boolean isHiddenDirectoryToSkip = isHidden(path) && !isChildOrGrandChildOfRoot(path);
+
+    if (isHiddenDirectoryToSkip || isExcludedDirectory(path) || isCoveredByExistingSources(path)) {
       return FileVisitResult.SKIP_SUBTREE;
     }
     return FileVisitResult.CONTINUE;
   }
 
-  private static boolean isHidden(Path path) {
-    return StreamSupport.stream(path.spliterator(), true)
-      .anyMatch(token -> token.toString().startsWith("."));
+  private boolean isHidden(Path path) {
+    Path relativePath = root.relativize(path);
+    while (relativePath != null && !relativePath.equals(root)) {
+      if (relativePath.toFile().isHidden()) {
```

Comment:
No need to convert the path to call `relativePath.toFile().isHidden()` you can do it using `Files.isHidden(path)`
But why changing `.startsWith(".")` by `.isHidden(path)`?
I don't know if it is reliable. If I extract a GitHub repo on a Linux and a Windows machine, will both OSs consider the same files hidden?